prow/config/cache.go

```diff
@@ -20,14 +20,76 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"strings"
 	"time"
 
+	"github.com/hashicorp/golang-lru/simplelru"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/sirupsen/logrus"
 
 	"k8s.io/test-infra/prow/cache"
 	"k8s.io/test-infra/prow/git/v2"
 )
 
+// MkEvictCallback constructs a simplelru.EvictCallback function to be used with
+// an InRepoConfigCache. It requires the name of the component and the evictions
+// metric to be used for the callback.
+func MkEvictCallback(component string) simplelru.EvictCallback {
```

Comment:
Lets just make this part of the LRU cache so we don't have to add it to every `main.go`. It looks like the only reason we're doing that now is to get the component name, but that is already globally available via `version.Name`.
https://github.com/kubernetes/test-infra/blob/688ff4ad2f85e5af9da0d2698b5dc42936756787/prow/version/doc.go#L28

Reply:
FTR we don't want to add the component name as a label because we already get that "for free" via `resource.container_name`. And, we probably don't want that anyway because we want to keep metrics for multiple deployments of the same component separated.